src/SmartTransactionsController.ts

```diff
@@ -103,6 +109,7 @@ export default class SmartTransactionsController extends BaseController<
       provider: any;
```

Comment:
```suggestion
      provider: Provider;
```

This can be done without disrupting anything, since the errors caused by this are already suppressed by `// @ts-expect-error TODO: Provider type alignment` directives.

Reply:
done here: https://github.com/MetaMask/smart-transactions-controller/pull/237/commits/93a05f14cbf91a344332dcf07d39f6924d5eacd3